src/Sylius/Component/Core/Factory/ChannelFactoryInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Factory;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+interface ChannelFactoryInterface extends FactoryInterface
```

Comment:
Is this interface even needed, as we have the interface from component namespace? https://github.com/Sylius/Sylius/blob/7901f9a3c65f626a6b40fa9b7ffa8c3a2df18ab3/src/Sylius/Component/Channel/Factory/ChannelFactoryInterface.php#L21

If so, it should be extended rather than replaced.

Reply:
IMHO, we need it while we don't have the covariance provided by PHP 7.4.

Reply:
However, all of the current implementations, that are based on the previous interface cannot be changed, due to BC Policy. If we want to extend the previous implementation we need to extend the previous interface